dbt/adapters/bigquery/python_submissions.py

```diff
@@ -118,16 +120,23 @@ def _get_job_client(self) -> dataproc_v1.BatchControllerClient:
     def _submit_dataproc_job(self) -> dataproc_v1.types.jobs.Job:
         batch = self._configure_batch()
         parent = f"projects/{self.credential.execution_project}/locations/{self.credential.dataproc_region}"
+        batch_id = uuid.uuid4().hex
 
-        request = dataproc_v1.CreateBatchRequest(
-            parent=parent,
-            batch=batch,
-        )
+        request = dataproc_v1.CreateBatchRequest(parent=parent, batch=batch, batch_id=batch_id)
         # make the request
-        operation = self.job_client.create_batch(request=request)  # type: ignore
+        self.job_client.create_batch(request=request)
         # this takes quite a while, waiting on GCP response to resolve
         # (not a google-api-core issue, more likely a dataproc serverless issue)
-        response = operation.result(polling=self.result_polling_policy)
+
+        state = "State.PENDING"
+        while state not in ["State.SUCCEEDED", "State.FAILED", "State.CANCELLED"]:
```

Comment:
I think we can save the last `time.sleep(2)` and some compute time if we do the first run outside and the while loop. Here's what I'm thinking without actually testing that it works:

```python
batch_request = dataproc_v1.GetBatchRequest(
    name=f"{parent}/batches/{batchid}"
)
end_states = [
    "State.SUCCEEDED",
    "State.FAILED",
    "State.CANCELLED"
]

response = self.job_client.get_batch(request=batch_request)
while str(response.state) not in end_states
    time.sleep(2)
    response = self.job_client.get_batch(request=batch_request)

return response
```

It may not seem like a lot of time for one use, but if this gets run many times, it could add up. Open to other options as well of course.

Reply:
I moved the sleep before the request to eliminate the last 2 seconds sleep.